src/main.cc

```diff
@@ -178,6 +178,39 @@ signal_cb(ev_loop *loop, struct ev_signal *w, int revents)
 	tarantool_exit(0);
 }
 
+static void
+signal_sigint_cb(ev_loop *loop, struct ev_signal *w, int revents)
+{
+	(void) loop;
+	(void) w;
+	(void) revents;
+
+	/**
+	 * If running in daemon mode, tarantool exits on SIGINT.
+	 * Complain about possibly sudden and unexpected death.
+	 */
+	if (pid_file) {
+		say_crit("got signal %d - %s", w->signum, strsignal(w->signum));
+		tarantool_exit(0);
+	}
+
+	/**
+	 * Setting prompt explicitly every time, cause of
+	 * need to return from search mode
+	 */
+	rl_set_prompt("tarantool> ");
+
+	const char *line_end = "^C\n";
+	ssize_t rc = write(STDOUT_FILENO, line_end, strlen(line_end));
+	if (rc < 0)
+		(void) rc;
```

Comment:
Just

```c
ssize_t rc = write(<...>);
(void)rc;
```

...is enough. It is just the hint for a compiler, we do nothing here in runtime.

Reply:
Seems resolved now, closing the comment.